### PR TITLE
Expose sqlalchemy pool recycle option for mysql/postgres

### PR DESCRIPTION
**Pull request by adam-bloom**
### Summary & Motivation
We experience intermittent dagit postgres connection issues (similar to #8462 and #8877).

This PR exposes a new parameter to the dagit invocation that is passed to sqlalchemy for mysql and postgres connections when in dagit mode for `pool_recycle`.

I put a little thought into making this just be `extra_kwargs` or something, but I couldn't think of a non-clunky way to pass that via a CLI arg (string encoded json?)

### How I Tested These Changes
Updated unit tests.

### CODE REVIEW

File: python_modules/dagit/dagit/cli.py [hunk 1]
```diff
@@ -86,6 +87,13 @@ def create_dagit_cli():
     type=click.INT,
     show_default=True,
 )
+@click.option(
+    "--db-pool-recycle",
+    help="The maximum age of a connection to use from the sqlalchemy pool without connection recycling. Set to -1 to disable. Not respected in all configurations.",
```
Comment (alangenfeld) [inline python_modules/dagit/dagit/cli.py:92]: > Set to -1 to disable

is this how it works at the sqlalchemy level?

**Review by alangenfeld — COMMENTED** [review]
awesome! thanks for taking the time to do this. Running CI now

File: python_modules/dagit/dagit/cli.py (hunk 1 above)
Comment (adam-bloom) [reply python_modules/dagit/dagit/cli.py:92]: Yup.

> this setting causes the pool to recycle connections after the given number of seconds has passed. It defaults to -1, or no timeout.

**Comment by alangenfeld** [comment]
seeing one issue in the helm json schema test - looks like casing got changed but the file not regenerated?
```
[2022-11-08T15:18:46Z]                  "dbPoolRecycle": {
--
  | [2022-11-08T15:18:46Z] -                    "title": "DbPoolRecycle",
  | [2022-11-08T15:18:46Z] +                    "title": "Dbpoolrecycle",

```

**Comment by adam-bloom** [comment]
> think you just need to re-run `dagster-helm schema apply`

Yes indeed, fixed!

**Pull request merged**